src/SDK/Resource/ResourceInfo.php

```diff
@@ -122,4 +125,25 @@ public function getSchemaUrl(): ?string
     {
         return $this->schemaUrl;
     }
+
+    private function mergeWith(ResourceInfo $resource): ResourceInfo
```

Comment:
Please make this method public and add it to the interface, since static `merge`is subject to be moved.